### PR TITLE
Refactored Bill

### PR DESCRIPTION
**Pull request by eriklsy**
Changed types within resolvers

### CODE REVIEW

**Comment by jaykbpark** [comment]
I wouldn't really call this refactoring -- please add better comments and titles in the future to avoid confusion.

File: backend/src/resolvers/bill.ts [hunk 1]
```diff
@@ -1,22 +1,22 @@
 import { get } from 'http';
```
Comment (cameronl10) [inline backend/src/resolvers/bill.ts]: can u delete this pls

File: backend/src/handlers/billHandler.ts [hunk 2]
```diff
@@ -78,7 +78,7 @@ async function DeleteBill(billid: String): Promise<void> {
 }
 
 //Gets single bill by billid
-async function GetBill(billid: string): Promise<Bill> { 
+async function GetBill(billid: String): Promise<Bill> { 
     const client = await Pool.connect();
     try {
         const result = await client.query('SELECT * FROM bill WHERE billid = $1', [billid]);
```
Comment (jaykbpark) [inline backend/src/handlers/billHandler.ts:84]: What is this for?

File: backend/src/resolvers/bill.ts (hunk 1 above)
Comment (eriklsy) [reply backend/src/resolvers/bill.ts]: deleted

File: backend/src/handlers/billHandler.ts (hunk 2 above)
Comment (eriklsy) [reply backend/src/handlers/billHandler.ts:84]: This is for Allocation, to get its parent

File: backend/src/resolvers/bill.ts (hunk 1 above)
Comment (jaykbpark) [inline backend/src/resolvers/bill.ts]: This get is never used no?

File: backend/src/resolvers/bill.ts [hunk 3]
```diff
@@ -1,22 +1,22 @@
...
-import { CreateBill, DeleteBill, EditBill, GetBills } from '../handlers/billHandler';
+import * as billHandler from '../handlers/billHandler';
 import { GetAllocations } from '../handlers/allocationHandler';
 
 export const resolvers = {
     Query: {
-        bills: async (_: any, { houseid }: any) => {
-            return await GetBills(houseid);
+        bills: async (_: any, { houseid }: {houseid: String}) => {
```
Comment (jaykbpark) [inline backend/src/resolvers/bill.ts]: I think typescript string is lower case

File: backend/src/resolvers/bill.ts [hunk 4]
```diff
@@ -1,22 +1,22 @@
...
+            return await billHandler.CreateBill(bill);
         },
-        editBill: async (_: any, { bill }: any): Promise<void> => {
-            return await EditBill(bill);
+        editBill: async (_: any, { bill }: {bill: Bill}): Promise<void> => {
+            return await billHandler.EditBill(bill);
         },
-        deleteBill: async (_: any, { billid }: any): Promise<void> => {
-            return await DeleteBill(billid);
+        deleteBill: async (_: any, { billid }: {billid: String}): Promise<void> => {
```
Comment (jaykbpark) [inline backend/src/resolvers/bill.ts]: I think typescript string is lower case

File: backend/src/handlers/billHandler.ts [hunk 5]
```diff
@@ -78,7 +78,7 @@ async function DeleteBill(billid: String): Promise<void> {
 }
 
 //Gets single bill by billid
-async function GetBill(billid: string): Promise<Bill> { 
+async function GetBill(billid: String): Promise<Bill> {
```
Comment (jaykbpark) [inline backend/src/handlers/billHandler.ts]: I think typescript string is lower case

**Comment by eriklsy** [comment]
> I wouldn't really call this refactoring -- please add better comments and titles in the future to avoid confusion.

I am deeply sorry I will keep this in mind. Please bear with me

File: backend/src/resolvers/bill.ts [hunk 6]
```diff
@@ -1,22 +1,22 @@
...
-import { CreateBill, DeleteBill, EditBill, GetBills } from '../handlers/billHandler';
+import * as billHandler from '../handlers/billHandler';
 import { GetAllocations } from '../handlers/allocationHandler';
 
 export const resolvers = {
     Query: {
-        bills: async (_: any, { houseid }: any) => {
-            return await GetBills(houseid);
+        bills: async (_: any, { houseid }: {houseid: String}) => {
+            return await billHandler.GetBills(houseid);
```
Comment (jaykbpark) [inline backend/src/resolvers/bill.ts]: The await keyword is unnecessary when directly returning a promise, you can remove await and just return the promise.
 
 This applies to all resolvers

File: backend/src/handlers/billHandler.ts [hunk 7]
```diff
@@ -1,6 +1,6 @@
 import Pool from '../../config/dbConnect';
 // Get all bills by houseid
-async function GetBills(houseid: string): Promise<Bill[]> {
+async function GetBills(houseid: String): Promise<Bill[]> {
```
Comment (jaykbpark) [inline backend/src/handlers/billHandler.ts]: can you make all functions camelcase?

File: backend/src/resolvers/bill.ts [hunk 8]
```diff
@@ -1,22 +1,22 @@
...
         }
     },
     Mutation: {
-        createBill: async (_: any, { bill }: any): Promise<Bill> => {
-            return await CreateBill(bill);
+        createBill: async (_: any, { bill }: {bill: Bill}): Promise<Bill> => {
```
Comment (jaykbpark) [inline backend/src/resolvers/bill.ts:11]: nit -- the  _: any parameter: is used to ignore the first parameter in GraphQL resolvers, but you might want to explicitly type it as parent: any for clarity.

**Review by jaykbpark — CHANGES_REQUESTED** [review]
please review & create changes

File: backend/src/resolvers/bill.ts (hunk 1 above)
Comment (eriklsy) [reply backend/src/resolvers/bill.ts]: cameron said to delete 👍 it was deleted in the new push

**Review by cameronl10 — APPROVED** [review]
lgtm
ACD_Systems_Logo

**Comment by jaykbpark** [comment]
looks better to me now.

File: backend/src/handlers/allocationHandler.ts [hunk 9]
```diff
@@ -76,7 +76,7 @@ async function DeleteAllocation(billid: String, userid: String): Promise<void> {
 };
 
 // Owed user is the user that is owed money, not the user that owes money
-async function GetAllocationOwed(userId: String, owedUserid: String): Promise<Number> {
+async function getAllocationOwed(userId: String, owedUserid: String): Promise<number> {
```
Comment (jaykbpark) [inline backend/src/handlers/allocationHandler.ts]: string?

File: backend/src/handlers/allocationHandler.ts [hunk 10]
```diff
@@ -64,7 +64,7 @@ async function EditAllocation(allocation: Partial<Allocation>): Promise<Allocati
     }
 };
 
-async function DeleteAllocation(billid: String, userid: String): Promise<void> {
+async function deleteAllocation(billid: String, userid: String): Promise<void> {
```
Comment (jaykbpark) [inline backend/src/handlers/allocationHandler.ts]: string?

File: backend/src/handlers/allocationHandler.ts [hunk 11]
```diff
@@ -1,6 +1,6 @@
 import Pool from '../../config/dbConnect';
 
-async function GetAllocations(billid: String): Promise<Allocation[]> {
+async function getAllocations(billid: String): Promise<Allocation[]> {
```
Comment (jaykbpark) [inline backend/src/handlers/allocationHandler.ts]: string?

**Comment by jaykbpark** [comment]
none of the [backend/src/handlers/allocationHandler.ts]
re-throws errors @cameronl10

**Review by jaykbpark — DISMISSED** [review]
please also check why none of our resolvers re-throw errors?

**Review by cameronl10 — APPROVED** [review]
nice

**Pull request merged**